### PR TITLE
Introduce DFUDevice, DFUDriver

### PR DESCRIPTION
**Pull request by Bastian-Krause**
**Description**
Add a resource, driver and client sub command to communicate with devices in DFU (Device Firmware Upgrade) mode. labgrid uses the [dfu-util tool](http://dfu-util.sourceforge.net/).

**Checklist**
- [x] Documentation for the feature
- [ ] Tests for the feature 
- [x] The arguments and description in doc/configuration.rst have been updated
- [x] CHANGES.rst has been updated
- [x] PR has been tested
- [x] Man pages have been regenerated

### CODE REVIEW

File: doc/configuration.rst [hunk 1]
```diff
@@ -1573,6 +1573,25 @@ Arguments:
     same as the fastboot default, which is computed after querying the target's
     ``max-download-size`` variable.
 
+DFUDriver
+~~~~~~~~~
+A DFUDriver allows the upload of images to a device in DFU (Device Firmware
```
Comment (jluebbe) [inline doc/configuration.rst:1578]: s/download/upload/

Comment (Bastian-Krause) [reply doc/configuration.rst:1578]: Right, I got confused by the upload/download terms in dfu-util. I'll fix it.

Comment (Bastian-Krause) [reply doc/configuration.rst]: Fixed.

**Comment by Bastian-Krause** [comment]
- fixed upload/download confusion in `DFUDriver` docs
- rebased on master

**Review by jluebbe — APPROVED** [review]

**Comment by jluebbe** [comment]
Thanks!

**Pull request merged**